sdks/python/apache_beam/transforms/enrichment_handlers/vertex_ai_feature_store.py

```diff
@@ -0,0 +1,289 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+from typing import List
+
+import proto
+from google.api_core.exceptions import NotFound
+from google.cloud import aiplatform
+
+import apache_beam as beam
+from apache_beam.transforms.enrichment import EnrichmentSourceHandler
+from apache_beam.transforms.enrichment_handlers.utils import ExceptionLevel
+
+__all__ = [
+    'VertexAIFeatureStoreEnrichmentHandler',
+    'VertexAIFeatureStoreLegacyEnrichmentHandler'
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _not_found_err_message(
+    feature_store_name: str, feature_view_name: str, entity_id: str) -> str:
+  """returns a string formatted with given parameters"""
+  return (
+      "make sure the Feature Store: %s with Feature View "
+      "%s has entity_id: %s" %
+      (feature_store_name, feature_view_name, entity_id))
+
+
+class VertexAIFeatureStoreEnrichmentHandler(EnrichmentSourceHandler[beam.Row,
+                                                                    beam.Row]):
+  """Enrichment handler to interact with Vertex AI Feature Store.
+
+  Use this handler with :class:`apache_beam.transforms.enrichment.Enrichment`
+  transform when the Vertex AI Feature Store is set up for
+  Bigtable Online serving.
+
+  With the Bigtable Online serving approach, the client fetches all the
+  available features for an entity-id. To filter the features to enrich, use
+  the `join_fn` param in :class:`apache_beam.transforms.enrichment.Enrichment`.
+
+  **NOTE:** The default severity to report exceptions is logging a warning. For
+    this handler, Vertex AI client returns the same exception
+    `Requested entity was not found` even though the feature store doesn't
+    exist. So make sure the feature store instance exists or set
+    `exception_level` as `ExceptionLevel.RAISE`.
```

Comment:
Rather than having this here, could we check this as part of `__enter__` and raise the exception no matter what there?

Reply:
This is possible with the Legacy feature store approach. But with the online instance method, I didn't find a method to check if the store instance exists except to make a call. And it returns the same error for invalid feature store name as that of the invalid entity-id name

Reply:
As discussed offline, I think we should be able to do this with a GET call to list available features, which should return some sort of not found error. I'm content leaving this to a future PR though, could you please just track that as an issue?

Reply:
Sounds good, I'll add that as TODO and get this in for 2.55.0 for now